playwright/pages/dialogs/addPromotionRuleDialog.ts

```diff
@@ -0,0 +1,70 @@
+import type { Page } from "@playwright/test";
+
+export class AddPromotionRuleDialog {
+    readonly page: Page;
+    constructor(
+        page: Page,
+        readonly ruleNameInput = page.getByTestId("rule-name-input"),
+        readonly ruleChannelDropdown = page.getByTestId("channel-dropdown"),
+        readonly selectOption = page.getByTestId(
+            "select-option",
+        ),
+        readonly ruleDescriptionInput = page.getByTestId("rich-text-editor-description"),
+        readonly addRuleConditionButton = page.getByTestId("add-rule-condition-button"),
+        readonly addRuleConditionSection = page.getByTestId("conditions-section"),
+        readonly addRuleConditionNameInput = page.getByTestId("condition-name-0"),
+        readonly addRuleConditionValueInput = page.getByTestId("condition-value-0"),
+        readonly addRuleConditionTypeDropdown = page.getByTestId("condition-type-0"),
+        readonly conditionRemoveButton = page.getByTestId("condition-remove-0"),
+        readonly rewardTypeSelect = page.getByTestId("reward-type-select"),
+        readonly percentageRewardValueTypeOption = page.getByTestId("percentage-reward-value-type"),
+        readonly fixedRewardValueTypeOption = page.getByTestId("fixed-reward-value-type"),
+        readonly rewardGiftSelect = page.getByTestId("reward-gifts-select"),
+    ) {
+        this.page = page;
+    }
+    async typePromotionRuleDescription(description: string) {
+        await this.ruleDescriptionInput.fill(description);
+    }
+    async clickChannelsDropdown() {
+        await this.ruleChannelDropdown.click();
+    }
+    async selectSingleChannel(channel = "PLN") {
+        await this.selectOption.filter({ hasText: `Channel-${channel}` }).click();
```

Comment:
question: `selectOption` exists only when the dropdown is open, how does it work that you click on it before open dropdown?
Also I see that method `selectSingleChannel` is not used

Reply:
@poulch I've removed `addPromotionRuleDialog.ts` file for now as it is not needed until I create tests for promotion rules (will be added back in the next PR).